ports/unix/main.c

```diff
@@ -453,10 +453,13 @@ static void set_sys_argv(char *argv[], int argc, int start_arg) {
 
 #if MICROPY_PY_SYS_EXECUTABLE
 extern mp_obj_str_t mp_sys_executable_obj;
-static char executable_path[MICROPY_ALLOC_PATH_MAX];
+static char *executable_path = NULL;
 
 static void sys_set_excecutable(char *argv0) {
-    if (realpath(argv0, executable_path)) {
+    if (executable_path == NULL) {
+        executable_path = realpath(argv0, NULL);
+    }
+    if (executable_path != NULL) {
```

Comment:
The memory allocated for `executable_path` is never freed.  That's usually not a big problem, but in debug builds the code tries to free all memory before the program finishes, to help find memory leaks (see the end of `main_`).

So I suggest to either:
1. free `executable_path` at the end of `main_` when `!defined(NDEBUG)`; or
2. make `executable_path` a local variable in this function and free it at the end, and take a copy of the string on the MicroPython heap (`m_new(char, strlen(executable_path) + 1)` etc) to pass into `mp_obj_str_set_data`